core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -103,7 +104,9 @@ object DynamicBrokerConfig {
         List(KafkaConfig.LogFlushIntervalMsProp, KafkaConfig.LogFlushSchedulerIntervalMsProp)
       case KafkaConfig.LogRetentionTimeMillisProp | KafkaConfig.LogRetentionTimeMinutesProp | KafkaConfig.LogRetentionTimeHoursProp =>
         List(KafkaConfig.LogRetentionTimeMillisProp, KafkaConfig.LogRetentionTimeMinutesProp, KafkaConfig.LogRetentionTimeHoursProp)
-      case ListenerConfigRegex(baseName) if matchListenerOverride => List(name, baseName)
+      case ListenerConfigRegex(baseName) if matchListenerOverride =>
+        val mechanismConfig = ListenerMechanismConfigs.find(baseName.endsWith)
+        List(name, mechanismConfig.getOrElse(baseName))
```

Comment:
Could we have a simple test case to cover this change? Also, I'm puzzling a little over why we do not include the `baseName` in all cases.

Reply:
If `baseName` is `plain.sasl.jaas.config`, we want to add `sasl.jaas.config` as a synonym. If `baseName` is `ssl.keystore.password`, we want to add the `baseName`. Added a comment and a unit test.